### DISCUSSION TITLE
Help building a simple interactive TUI

### DISCUSSION THREAD

**Discussion by jeffwright13** [Textualize/textual#231] [Q&A]
Hello, I am interested in using Textual to build a simple TUI for a project I am working on. What I would like to do is have the ability to click some part of a text widget so that it "folds" (hides/disappears) when clicked, and then "unfolds" (unhides/re-appears) when clicked again, displaying its text anew. I had originally used Rich to build up a Console instance with multiple Panels to display the text sections, but I haven't been able to figure out how to handle mouse input. `pynput` seems too rudimentary, as all it gives me is the absolute coordinates of a click, so I don't know how to tell if the mouse click is clicking on one widget or the other. 

I have looked through the example files. Not having a lot of experience with reactive web design or anyscio, it isn't 100% clear how they all work. Can anyone give me a push in the right direction?

**Comment by evglitvin** [comment]
Here in this repo the file 
https://github.com/Textualize/textual/blob/main/src/textual/drivers/linux_driver.py

in that file u can find almost everything as keyinput as well as setup terminal to work without echoing etc

reading mouse itself u can found in 
https://github.com/Textualize/textual/blob/main/src/textual/_xterm_parser.py

thats basically everything u need
